var/spack/repos/builtin/packages/python/package.py

```diff
@@ -196,6 +196,114 @@ class Python(AutotoolsPackage):
     # An in-source build with --enable-optimizations fails for python@3.X
     build_directory = 'spack-build'
 
+    executables = ['python']
+
+    @classmethod
+    def determine_spec_details(cls, prefix, exes_in_prefix):
+        """Allow ``spack external find python`` to locate Python installations.
+
+        Parameters:
+            prefix (str): the directory containing the executables
+            exes_in_prefix (set): the executables that match the regex
+
+        Returns:
+            spack.spec.Spec: the spec of this Python installation
+        """
+        # Possible executable names:
+        # * python, python3, python3.7, python3.7m
+        # Possible red herrings:
+        # * python-config, python3-config, python3.7-config
+        # * spack-python
+        # Take the shortest executable name and hope for the best
+        python = Executable(min(exes_in_prefix))
+        if python.name.startswith('spack-') or python.name.endswith('-config'):
+            return None
+
+        # First, get the Python version
+        # Newer versions of Python support `--version`,
+        # but older versions only support `-V`
+        # Python 2 sends to STDERR, while Python 3 sends to STDOUT
+        # Output looks like:
+        #   Python 3.7.7
+        output = python('-VV', output=str, error=str)
+        match = re.match(r'Python\s+(\S+)', output)
+        version = ''
+        if match:
+            version = '@' + match.group(1)
+
+        # The use of `-VV` instead of `-V` above gives additional information,
+        # including which compiler was used to build the executable, at least
+        # for Python 3
+        # Output looks like:
+        #   Python 3.7.7 (default, May  5 2020, 22:46:17)
+        #   [Clang 11.0.3 (clang-1103.0.32.59)]
+        # TODO: How to make this more robust? How to differentiate between
+        # Clang and Apple Clang?
+        match = re.search(r'(Clang|GCC)\s+(\S+)', output)
+        compiler = ''
+        if match:
+            compiler = '%' + match.group(1).lower() + '@' + match.group(2)
```

Comment:
This is the buggiest part of the code. It doesn't correctly detect Apple Clang, and I don't have access to Python built with any other compilers, so I can't test anything other than Apple Clang. I'm tempted to just leave this out.

Reply:
You might be able to get this information from `python -c "from __future__ import print_function; import sys;  print(sys.implementation._multiarch)"`or a similar call, but I don't have access to Apple Clang.